File: LeetCode Solutions/Study Plan/Programming Skills/Programming Skills II/Monotonic Array.c


bool isMonotonic(int* nums, int numsSize){
    bool increasing = false, decreasing = false;
    for (int i=0; i<numsSize-1; i++){
        if (nums[i+1] > nums[i]){
            increasing = true;
        }
        if (nums[i+1] < nums[i]){
            decreasing = true;
        }
        if (increasing == true && decreasing == true){
            return false;
        }
    }
    return true;
}

File: LeetCode Solutions/Study Plan/Programming Skills/Programming Skills II/Monotonic Array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        increasing = decreasing = False
        for i in range(len(nums)-1):
            if nums[i+1]>nums[i]:
                increasing = True
            if nums[i+1] < nums[i]:
                decreasing = True
            if increasing == decreasing == True:
                return False
        return True